### ISSUE TITLE
You are using a model of type deepseek_vl_v2 to instantiate a model of type DeepseekOCR. This is not supported for all configurations of models and can yield errors.

### ISSUE THREAD

**Issue by jinxiu0406** [oomol-lab/pdf-craft#334]
环境：
windows11
python 3.12
torch 2.6.0+cu126
pdf-craft 1.08

代码:
`
from pdf_craft import transform_markdown

transform_markdown(
    pdf_path="E:/books/33趣味离散数学.pdf",
    markdown_path="33趣味离散数学.md",
    local_only=True,  # 仅使用本地模型，不从网络下载
    pdf_handler=DefaultPDFHandler(poppler_path="E:/soft/poppler-25.12.0/Library/bin"),
    markdown_assets_path="images",
    analysing_path="temp",  # 可选：指定临时文件夹
    ocr_size="gundam",  # 可选：tiny, small, base, large, gundam
    models_cache_path="./my_models",  # 可选：模型缓存路径
    dpi=300,  # 可选：渲染 PDF 页面的 DPI（默认：300）
    max_page_image_file_size=None,  # 可选：最大图像文件大小（字节），超出时自动调整 DPI
    includes_footnotes=True,  # 可选：包含脚注
    ignore_pdf_errors=False,  # 可选：遇到 PDF 渲染错误时继续处理
    ignore_ocr_errors=False,  # 可选：遇到 OCR 识别错误时继续处理
)`

cmd 信息:

You are using a model of type **deepseek_vl_v2** to instantiate a model of type **DeepseekOCR**. This is not supported for all configurations of models and can yield errors.  
Some weights of DeepseekOCRForCausalLM were not initialized from the model checkpoint at \\192.168.xx.xx\x\xx\xxxxxx\LLM_knowleage\my_models\models--deepseek-ai--DeepSeek-OCR\snapshots\9f30c71f441d010e5429c532364a86705536c53a and are newly initialized: ['model.vision_model.embeddings.position_ids']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
C:\Users\xxxxxx\AppData\Roaming\Python\Python312\site-packages\transformers\generation\configuration_utils.py:628: UserWarning: `do_sample` is set to `False`. However, `temperature` is set to `0.0` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `temperature`.
  warnings.warn(
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:1 for open-end generation.
The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
The `seen_tokens` attribute is deprecated and will be removed in v4.41. Use the `cache_position` model input instead.
`get_max_cache()` is deprecated for all Cache classes. Use `get_max_cache_shape()` instead. Calling `get_max_cache()` will raise error from v4.48
=====================
BASE:  torch.Size([1, 256, 1280])
PATCHES:  torch.Size([6, 100, 1280])
=====================
The attention layers in this model are transitioning from computing the RoPE embeddings internally through `position_ids` (2D tensor with the indexes of the tokens), to using externally computed `position_embeddings` (Tuple of tensors, containing cos and sin). In v4.46 `position_ids` will be removed and `position_embeddings` will be mandatory.
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:1 for open-end generation.
=====================
BASE:  torch.Size([1, 256, 1280])
PATCHES:  torch.Size([6, 100, 1280])
=====================
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:1 for open-end generation.
=====================
BASE:  torch.Size([1, 256, 1280])
PATCHES:  torch.Size([6, 100, 1280])
=====================
.........

**Comment by Moskize91** [comment]
失败了吗？看着像警告，可以不理它。

**Comment by jinxiu0406** [comment]
> 失败了吗？看着像警告，可以不理它。

md文件生成了。似乎没有影响。

**Comment by Moskize91** [comment]
能生成就行，说明没影响，不用管它。
